### ISSUE TITLE
ERROR: Postprocessing: ffmpeg not found. Please install or provide the path using --ffmpeg-location

### ISSUE THREAD

**Issue by Williamtung1** [jely2002/youtube-dl-gui#359]
**Describe the bug**
encounter "ERROR: Postprocessing: ffmpeg not found. Please install or provide the path using --ffmpeg-location"
the video and audio downloaded are separated

**To Reproduce**
1. download the video and audio from https://www.youtube.com/watch?v=rfscVS0vtbw
2. Click on download
3. encounter 'Error! unhandle error'
4. The full error "ERROR: Postprocessing: ffmpeg not found. Please install or provide the path using --ffmpeg-location"

**Expected behavior**
I expected it should download and merge the video and audio together.

**Screenshots**

**Additional info (please complete the following information):**
 - OS: macOS
 - Application version: 2.4.0
 - Application type: dmg installer

**Additional context**
I do have latest version of ffmpeg downloaded and installed in my macOS

xxx@xxx-MacBook-Pro ~ % ffmpeg
ffmpeg version 5.0.1 Copyright (c) 2000-2022 the FFmpeg developers
  built with Apple clang version 13.0.0 (clang-1300.0.29.30)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/5.0.1 --enable-shared --enable-pthreads --enable-version3 --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libbluray --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-librav1e --enable-librist --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libsoxr --enable-libzmq --enable-libzimg --disable-libjack --disable-indev=jack --enable-videotoolbox
  libavutil      57. 17.100 / 57. 17.100
  libavcodec     59. 18.100 / 59. 18.100
  libavformat    59. 16.100 / 59. 16.100
  libavdevice    59.  4.100 / 59.  4.100
  libavfilter     8. 24.100 /  8. 24.100
  libswscale      6.  4.100 /  6.  4.100
  libswresample   4.  3.100 /  4.  3.100
  libpostproc    56.  3.100 / 56.  3.100
Hyper fast Audio and Video encoder
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...

**Comment by i-Captain** [comment]
I ran into the same error (OSX, Version: 2.4.0). I found the fix at stackoverflow => `brew install youtube-dl`.

After that, the video download (output .mp4) is working, but the audio is 'missing'. Even after I removed youtube-dl and installed yt-dlp (looked at Changelog for v2.4.0).

I tried to download only the audio as .m4a and got an error, that AtomicParsley is not  installed. Looked at the source code and found 4 search results with 'AtomicParsley.exe'....

The workaround for me: 
Settings: Video output as .mkv 
Afterwards you can use ffmpeg to convert the video as .mp4 `ffmpeg -i video.mkv video.mp4`

For 'big' videos this will take a while...

So another option is to download the video without audio as .mp4 and then download only the audio as .opus

Convert the .opus into .m4a `ffmpeg -i audio.opus audio.m4a` and add the audio to the video `ffmpeg -i video.mp4 -i audio.m4a -map 0 -map 1 -c copy output.mp4`

**Comment by Williamtung1** [comment]
Yeah, thank you very much. I have also found the solution at github(using output=.mkv) But for me, it doesn't really matter.

**Comment by himyjan** [comment]
https://github.com/jely2002/youtube-dl-gui/issues/354#issuecomment-1127269708

**Comment by justwes** [comment]
Same error as reported by @Williamtung1 

Currently using a newer version of ffmpeg and I have YouTube-dl already installed via Homebrew.

Cannot find a method to update the application or force it to look in the correct location for ffmpeg

$ brew info youtube-dl
==> youtube-dl: stable 2021.12.17 (bottled)
Download YouTube videos from the command-line
https://youtube-dl.org/
/opt/homebrew/Cellar/youtube-dl/2021.12.17 (2,241 files, 18.3MB) *
  Poured from bottle on 2022-10-08 at 10:18:59
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/youtube-dl.rb
License: Unlicense
==> Dependencies
Required: python@3.10 ✔
==> Caveats
Bash completion has been installed to:
  /opt/homebrew/etc/bash_completion.d
==> Analytics
install: 12,266 (30 days), 38,697 (90 days), 271,564 (365 days)
install-on-request: 12,129 (30 days), 38,247 (90 days), 263,495 (365 days)
build-error: 3 (30 days)

$ ffmpeg
ffmpeg version 5.1.2 Copyright (c) 2000-2022 the FFmpeg developers
  built with Apple clang version 14.0.0 (clang-1400.0.29.102)
  configuration: --prefix=/opt/homebrew/Cellar/ffmpeg/5.1.2 --enable-shared --enable-pthreads --enable-version3 --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libbluray --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-librav1e --enable-librist --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libsoxr --enable-libzmq --enable-libzimg --disable-libjack --disable-indev=jack --enable-videotoolbox --enable-neon
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
Hyper fast Audio and Video encoder
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...

**Comment by jely2002** [comment]
This should be fixed in v3 as it now properly installs and points to the ffmpeg binary,
